### ISSUE TITLE
Overlay while recording video?

### ISSUE THREAD

**Issue by developerandroid123** [CrazyOrr/FFmpegRecorder#27]
Can we add overlay like text,image or video  while recording ?

**Comment by CrazyOrr** [comment]
Yes, please refer to #11

**Comment by developerandroid123** [comment]
I am not able to get any hint about overlay from #11

**Comment by Jagdeep-Sing** [comment]
i also want overlay

**Comment by CrazyOrr** [comment]
You should search for FFmpeg feature for your need, like [overlay](https://ffmpeg.org/ffmpeg-filters.html#overlay-1).

**Comment by Jagdeep-Sing** [comment]
i read that documentation and code, now where should is i place it? in resumeRecording() function? im not getting it

**Comment by Jagdeep-Sing** [comment]
can anyone please help who successfully completed it?

**Comment by Jagdeep-Sing** [comment]
How do i implement following code in this project:

`drawtext="fontsize=15:fontfile=FreeSerif.ttf:text=LONG_LINE:y=h-line_h:x=-50*t"`

this is drawtext property of ffmpeg, how do i implement this?

**Comment by CrazyOrr** [comment]
Check out `FFmpegFrameFilter` used in `VideoRecordThread`, there are 'transpose' and 'crop' filters used in this project already as examples for you.

**Comment by Jagdeep-Sing** [comment]
Thanks for your response!!
I see in code, you apply filter like this : 

`String transpose = null; `
` transpose = "transpose=cclock"; //cclock is a filter`
`//then you add filter like this`
` filters.add(transpose);` 

Can I do same also for adding text? like :
 
`String drawtext = null;`
 `drawtext = "drawtext='fontsize=15:fontfile=FreeSerif.ttf:text=LONG_LINE:y=h-line_h:x=-50*t'";`
 `//then apply text like this`
 `filters.add(drawtext);`

**am I right? please help...**

**Comment by Jagdeep-Sing** [comment]
**UPDATE**
In `drawtext` documentation of ffmpeg they wrote that : 

> To enable compilation of this filter, you need to configure FFmpeg with --enable-libfreetype.

**Comment by CrazyOrr** [comment]
Yes, according to [this](https://github.com/bytedeco/javacpp-presets/blob/master/ffmpeg/cppbuild.sh), `--enable-libfreetype` has already been configured for building this FFmpeg library. If not so, try again with the latest version of [JavaCV](https://github.com/bytedeco/javacv) or [build it by yourself](https://github.com/bytedeco/javacpp-presets/wiki/Build-Environments).

**Comment by Jagdeep-Sing** [comment]
okay thanks, can you then please help where i place this code, i tried with code that i posted here but it results in force close.

**Comment by CrazyOrr** [comment]
It crashes with `Fatal signal 11 (SIGSEGV), code 1, fault addr 0x18 in tid 24319` now. 
Refer to [this issue](https://github.com/bytedeco/javacv/issues/254) of JavaCV.

**Comment by Jagdeep-Sing** [comment]
Same error for me and I didn't find any solution there. that issue is for building JavaCV with libfreetype but this project is already configured for that, its a confusion now

**Comment by Jagdeep-Sing** [comment]
got same error when use :
`overlay`

**Comment by Jagdeep-Sing** [comment]
i think latest version of JAVACV can solve this. can you please guide me how do I use this ffmpeg project with latest code of JAVACV?
thanks for your continuous response!! :+1:

**Comment by Jagdeep-Sing** [comment]
UPDATE:
I tried it with latest version of JAVACV by changing the version in dependencies and got the same error.

**Comment by CrazyOrr** [comment]
You can try the commands out by using [FFmpeg](http://ffmpeg.org/ffmpeg.html) on your computer.

**Comment by Jagdeep-Sing** [comment]
yes in command line its working fine, I tested in Ubuntu
